### PR TITLE
Better croods typing

### PR DESCRIPTION
**Pull request by gustavoguichard**
Breaking changes:
- We are not exporting `CroodsContext` anymore (it was not even documented). But now we are exporting `useBaseOptions` hook which returns the options set on `CroodsProvider` and must be documented.
- We are not exporting `useCroodsStore` anymore as it was a broken API and not documented.
- No more proptypes in favor of full TS support.
- Fetch with no `info` nor any error or loading state used to return `null`, which was a bug. It should return what you ask for on `renderEmpty` or the result of `render` if you didn't provide the sooner.
- We are not exporting the undocumented actions `clearMessages` and `resetState` anymore

### CODE REVIEW

File: src/typeDeclarations.ts [hunk 1]
```diff
@@ -44,6 +52,7 @@ export interface ProviderOptions {
   debugActions?: boolean
   debugRequests?: boolean
   paramsParser?: (t: string) => string
+  parseParams?: (t: string) => string // TO REMOVE
```
Comment (angelod1as) [inline src/typeDeclarations.ts:55]: To remove? If this was intentional, discard this comment.

File: src/typeDeclarations.ts [hunk 2]
```diff
@@ -68,16 +77,16 @@ export interface InstanceOptions extends ProviderOptions {
...
   name: string
   stateId?: ID
   type?: 'list' | 'info'
   value: any
 }
 
 type HTTPMethod = 'POST' | 'PUT' | 'PATCH'
-
 export interface ActionOptions extends ProviderOptions {
+  requestConfig?: any // TO REMOVE
```
Comment (angelod1as) [inline src/typeDeclarations.ts:89]: same here

**Review by angelod1as — APPROVED** [review]

File: src/typeDeclarations.ts (hunk 1 above)
Comment (gustavoguichard) [reply src/typeDeclarations.ts]: Yes.. we'll remove it in the future bc it is an error that we can't remove without breaking the API.
We'll probably add deprecation warnings in the meantime...
Anyway, this PR is still a Draft, today I added a LOT more and still a draft :P

Comment (angelod1as) [reply src/typeDeclarations.ts]: I'll slap a WIP at the title then, to avoid further reviews while unfinished.

**Review by diogob — APPROVED** [review]
LGTM

**Review by giovanisleite — APPROVED** [review]

**Pull request merged**